pkg/scheduler/factory/plugins_test.go

```diff
@@ -136,3 +137,104 @@ func TestBuildScoringFunctionShapeFromRequestedToCapacityRatioArgumentsNilResour
 }
+
+func appendToPluginSet(pluginSet *config.PluginSet, name string) *config.PluginSet {
+	if pluginSet == nil {
+		pluginSet = &config.PluginSet{}
+	}
+	pluginSet.Enabled = append(pluginSet.Enabled, config.Plugin{Name: name})
+	return pluginSet
+}
+
+func TestRegisterFrameworkConfigProducers(t *testing.T) {
+	testPredicateName1 := "testPredicate1"
+	testFilterName1 := "testFilter1"
+	RegisterPredicateToFrameworkConfigProducer(testPredicateName1,
+		func() (plugins config.Plugins, pluginConfig []config.PluginConfig) {
+			plugins.Filter = appendToPluginSet(plugins.Filter, testFilterName1)
+			return
+		})
+
+	testPredicateName2 := "testPredicate2"
+	testFilterName2 := "testFilter2"
+	RegisterPredicateToFrameworkConfigProducer(testPredicateName2,
+		func() (plugins config.Plugins, pluginConfig []config.PluginConfig) {
+			plugins.Filter = appendToPluginSet(plugins.Filter, testFilterName2)
+			return
+		})
+
+	testPriorityName1 := "testPriority1"
+	testScoreName1 := "testScore1"
+	RegisterPriorityToFrameworkConfigProducer(testPriorityName1,
+		func() (plugins config.Plugins, pluginConfig []config.PluginConfig) {
+			plugins.Score = appendToPluginSet(plugins.Score, testScoreName1)
+			return
+		})
+
+	testPriorityName2 := "testPriority2"
+	testScoreName2 := "testScore2"
+	RegisterPriorityToFrameworkConfigProducer(testPriorityName2,
+		func() (plugins config.Plugins, pluginConfig []config.PluginConfig) {
+			plugins.Score = appendToPluginSet(plugins.Score, testScoreName2)
+			return
+		})
+
+	var allPlugins config.Plugins
+
+	// Verify that both predicates are in the map and produce the expected filter configurations.
+	predicates := []string{testPredicateName1, testPredicateName2}
+	filters := []string{testFilterName1, testFilterName2}
+	if len(predicateToFrameworkConfigProducer) != len(predicates) {
+		t.Errorf("len of predicateFrameworkConfigurator, got %v, wanted %v.",
+			len(predicateToFrameworkConfigProducer), len(predicates))
+	}
+	for _, predicate := range predicates {
+		producer, ok := predicateToFrameworkConfigProducer[predicate]
+		if !ok {
+			t.Fatalf("wanted to find %q in the configurator map.", predicate)
+		}
+		plugins, _ := producer()
+		allPlugins.Append(&plugins)
+	}
+
+	// Verify that both priorities are in the map and produce the expected score configurations.
+	priorities := []string{testPriorityName1, testPriorityName2}
+	scorers := []string{testScoreName1, testScoreName2}
+	if len(priorityToFrameworkConfigProducer) != len(priorities) {
+		t.Errorf("len of priorityToFrameworkConfigProducer, got %v, wanted %v.",
+			len(priorityToFrameworkConfigProducer), len(priorities))
+	}
+	for _, priority := range priorities {
+		producer, ok := priorityToFrameworkConfigProducer[priority]
+		if !ok {
+			t.Fatalf("wanted to find %q in the configurator map.", priority)
+		}
+		plugins, _ := producer()
+		allPlugins.Append(&plugins)
+	}
+
+	// Verify the aggregated configuration.
+	if len(allPlugins.Filter.Enabled) != len(predicates) {
```

Comment:
it's simpler if you define `wantPlugins` and then use `cmp.Diff`

Reply:
Done.